src/main/java/wiresegal/wob/arcanum/Parsing.kt

```diff
@@ -167,6 +173,78 @@ fun about(message: Message) {
     })
 }
 
+class Progress(title: String, percentage: String, link: String?) {
+    operator fun component1(): String {
+        return title
+    }
+    operator fun component2(): String {
+        return percentage
+    }
+    operator fun component3(): String? {
+        return link
+    }
+
+    private val title: String = title
+    private val percentage: String = percentage
+    private val link: String? = link
+}
+
+val progressCache = mutableListOf<Progress>()
+var progressCacheTimeStamp: Long? = null
+
+fun showProgressBar(message: Message) {
+    val progresses = extractProgresses()
+    val embed = EmbedBuilder ()
+            .setColor(embedColor)
+            .setTitle("Progress Bars")
+            .setUrl(homepageTarget)
+
+    val full = "█"
+    val empty = "░"
+
+    for (progress in progresses) {
+        val (name, percent, link) = progress
+        val percentNumber: Int = Integer.parseInt(percent)
+        val sensitivity = 20
+        val points: Int = (percentNumber/100.0 * sensitivity).toInt()
+        val bar = full.repeat(points) + empty.repeat(sensitivity - points)
+        val linkText = if(link != null) "_[check it out]($link)_\n" else ""
+        embed.addField("**$name**", "$linkText$bar $percent%", false).setUrl("https://google.com/")
```

Comment:
I assume you tested something here. I don't think we want to set Google as the URL on all these embeds :D